lldb/source/Breakpoint/Watchpoint.cpp

```diff
@@ -211,6 +212,35 @@ bool Watchpoint::CaptureWatchedValue(const ExecutionContext &exe_ctx) {
   return (m_new_value_sp && m_new_value_sp->GetError().Success());
 }
 
+bool Watchpoint::WatchedValueReportable(const ExecutionContext &exe_ctx) {
+  if (!m_watch_modify)
+    return true;
+  if (!m_type.IsValid())
+    return true;
+
+  ConstString watch_name("$__lldb__watch_value");
+  Address watch_address(GetLoadAddress());
+  ValueObjectSP newest_valueobj_sp = ValueObjectMemory::Create(
+      exe_ctx.GetBestExecutionContextScope(), watch_name.GetStringRef(),
+      watch_address, m_type);
+  newest_valueobj_sp = newest_valueobj_sp->CreateConstantValue(watch_name);
+  DataExtractor new_data;
+  DataExtractor old_data;
+  Status error;
+  newest_valueobj_sp->GetData(new_data, error);
+  m_new_value_sp->GetData(old_data, error);
+
```

Comment:
Do we need to update m_new_value_sp with the value in newst_valueobj_sp somewhere so we detect when things change again?

Reply:
In StopInfoWatchpoint::PerformAction() we decide if we're `m_should_stop` and if so, we call `Watchpoint::CaptureWatchedValue` (one of the things we do before that call is compare the value and decide if we're going to stop or not).  CaptureWatchedValue copies the "new" value into the m_old_value_sp and then creates a new constant value object for m_new_value_sp.  I copied that same way of creating the const ValueObject in this method; it's the method before this one in Watchpoint.cpp.  So the copying of the new value to the previous value is already handled by that code in CaptureWatchedValue.

Reply:
Should add, because I'm sure someone will wonder about this:  Yes, I'm creating a constant ValueObject with the watched memory here, and then throwing that ValueObject away.  And if we decide to stop, we'll create the constant ValueObject again with the same memory, to show it to the user.  I'm relying on the memory read cache to only read that memory from the inferior process once at a single stop, otherwise this double-creation of a ValueObject would be a real perf hit to watchpoints.